stock_archive_constraint/tests/test_location_archive_constraint.py

```diff
@@ -0,0 +1,338 @@
+        )
+        self.product_1.active = False
+        self.assertFalse(self.product_1.active)
+        self.company.active_stock_constraint = True
+        with self.assertRaises(ValidationError):
+            self.product_2.active = False
+
+    def test_archive_product_with_stock_picking_in_out(self):
+        self._create_stock_picking(
+            self.env.ref("stock.stock_location_suppliers"),
+            self.stock_location,
+            self.product_2,
+            20.00,
+        )
+        self._create_stock_picking(
+            self.stock_location,
+            self.env.ref("stock.stock_location_customers"),
+            self.product_2,
+            20.00,
+        )
+        self.product_1.active = False
+        self.assertFalse(self.product_1.active)
+        self.product_2.active = False
+        self.assertFalse(self.product_2.active)
+
+    def test_archive_product_stock_location(self):
+        self._create_stock_quant(self.stock_location, self.product_2, 20.00)
+        self.product_1.active = False
+        self.assertFalse(self.product_1.active)
+        self.company.active_stock_constraint = True
+        with self.assertRaises(ValidationError):
+            self.product_2.active = False
+
+    def test_archive_product_stock_location_child(self):
+        self._create_stock_quant(self.stock_location_child, self.product_2, 20.00)
+        self.product_1.active = False
+        self.assertFalse(self.product_1.active)
+        self.company.active_stock_constraint = True
+        with self.assertRaises(ValidationError):
+            self.product_2.active = False
+
+    def test_archive_unarchive_stock_location(self):
+        self.stock_location.active = False
+        self.assertFalse(self.stock_location.active)
+        self.stock_location.active = True
+        self.assertTrue(self.stock_location.active)
+
+    def test_archive_stock_location_ok(self):
+        self.stock_location.active = False
+        self.assertFalse(self.stock_location.active)
+
+    def test_archive_stock_location(self):
+        self._create_stock_quant(self.stock_location, self.product_2, 20.00)
+        self.company.active_stock_constraint = True
+        with self.assertRaises(ValidationError):
+            self.stock_location.with_context(do_not_check_quant=True).active = False
+
+    def test_archive_unarchive_stock_location_child(self):
+        self.stock_location_child.active = False
+        self.assertFalse(self.stock_location_child.active)
+        self.stock_location_child.active = True
+        self.assertTrue(self.stock_location_child.active)
+
+    def test_archive_stock_location_child(self):
+        self._create_stock_quant(self.stock_location_child, self.product_2, 20.00)
+        self.company.active_stock_constraint = True
+        with self.assertRaises(ValidationError):
+            self.stock_location.with_context(do_not_check_quant=True).active = False
+
+    def test_archive_product_from_ohter_company(self):
```

Comment:
typo 
```suggestion
    def test_archive_product_from_other_company(self):
```
fix also this typo in the other test